Client/Assets/SessionRestore.html

```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+ - License, v. 2.0. If a copy of the MPL was not distributed with this
+ - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+ <html>
+ <script>
+    (function () { 
+        var index = document.location.href.search("history");
+        var sessionRestoreComponents = JSON.parse(unescape(document.location.href.substring(index + "history=".length)))
```

Comment:
`// Pull the session out of the history query argument. The session is a JSON-stringified array of all URLs to restore for this tab, plus the last active index.`